api/src/services/cards.service.ts

```diff
@@ -11,15 +11,27 @@ import { ObjectId } from 'mongoose';
 
 @Service()
 export class CardService {
-  public async findAllCards(page: number, limit: number): Promise<PokemonCard[]> {
-    // Still need PokemonCardModel
-    const cards: PokemonCard[] = await PokemonCardModel.find()
-    // Ensure the limit is a number by multipling by 1
-    .limit(limit * 1)
-    // Skip the correct number of pages based on current page
-    // ie page 1 skip 0 cards
-    .skip((page - 1) * limit);
-    return cards;
+  public async findAllCards(page: number, limit: number, searchByName: boolean, cardName: string | undefined): Promise<PokemonCard[]> {
+
+    if(searchByName) {
+      const cards: PokemonCard[] = await PokemonCardModel.find({name: `${cardName}`})
+      // Ensure the limit is a number by multipling by 1
+      .limit(limit * 1)
+      // Skip the correct number of pages based on current page
+      // ie page 1 skip 0 cards
+      .skip((page - 1) * limit);
+      return cards;
+    }
+    else {
+      const cards: PokemonCard[] = await PokemonCardModel.find()
+      // Ensure the limit is a number by multipling by 1
+      .limit(limit * 1)
+      // Skip the correct number of pages based on current page
+      // ie page 1 skip 0 cards
+      .skip((page - 1) * limit);
+      return cards;
+    }
+    
```

Comment:
This looks good. But there may be a way to do this without rewriting, .limit().skip(); and return cards; So if you could think of a way to avoid that, that would be good but it does not need to be changed.

Reply:
I tried to do that, but it gave me a scope error on cards. To fix that, I tried declaring cards outside of the if statements, and initializing it inside the if statements. However, this would then not allow me to .limit or .skip on cards. It was really weird